prow/cmd/deck/main_test.go

```diff
@@ -87,18 +97,42 @@ func TestHandleLog(t *testing.T) {
 		if err != nil {
 			t.Fatalf("Error parsing URL: %v", err)
 		}
+		var follow = false
+		if ok, _ := strconv.ParseBool(u.Query().Get("follow")); ok {
+			follow = true
+		}
 		req.URL = u
 		rr := httptest.NewRecorder()
 		handler.ServeHTTP(rr, req)
 		if rr.Code != tc.code {
 			t.Errorf("Wrong error code. Got %v, want %v", rr.Code, tc.code)
 		} else if rr.Code == http.StatusOK {
-			resp := rr.Result()
-			defer resp.Body.Close()
-			if body, err := ioutil.ReadAll(resp.Body); err != nil {
-				t.Errorf("Error reading response body: %v", err)
-			} else if string(body) != "hello" {
-				t.Errorf("Unexpected body: got %s.", string(body))
+			if follow {
+				//wait a little to get the chunks
+				time.Sleep(2 * time.Millisecond)
```

Comment:
Why is this necessary? Shouldn't reader.readBytes('\n') block while waiting anyways?